### PR TITLE
Cleanup access levels in DataGridViewElement and subclasses

### PR DESCRIPTION
**Pull request by hughbe**
Prefer using `internal set` rather than `XInternal` properties

### CODE REVIEW

**Comment by weltkante** [comment]
Using internal accessors on properties will expose them more broadly than having a separate accessor/method. When using reflection you'll suddenly see the set-accessor appearing on the property unless you explicitely filter out the non-public accessor. When using databinding I'm not sure, ReflectPropertyDescriptor in [some](https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectPropertyDescriptor.cs#L389) code paths filters out non-public accessors, but in [others](https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectPropertyDescriptor.cs#L411) it doesn't.

Might not be a problem here but I thought I'd mention it anyways since I had a few problems of this kind in the past when refactoring my own code.

**Comment by RussKie** [comment]
We had a chat with @merriemcgaw and it was agreed as acceptable to amalgamte `internal setter only properties` into the `public` counterparts.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs [hunk 1]
```diff
@@ -462,44 +456,20 @@ public bool IsInEditMode
...
+        [Browsable(false)]
+        [EditorBrowsable(EditorBrowsableState.Advanced)]
         public DataGridViewColumn OwningColumn
         {
-            get
-            {
-                return owningColumn;
-            }
+            get => owningColumn;
+            set => owningColumn = value;
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewCell.cs]: TODO

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs [hunk 2]
```diff
@@ -30,7 +30,7 @@ public class DataGridViewBand : DataGridViewElement, ICloneable, IDisposable
         private int _thickness;
         private int _minimumThickness;
         private int _bandIndex;
-        internal bool _bandIsRow;
+        private protected bool _bandIsRow;
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs]: No sir - there would be a compile error if this was the case

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs]: I was just curious whether we need the `protected` accessor

[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs]: Oh, sorry! It is accessed from DataGridViewColumn and DataGridViewRow

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewBand.cs]: Thank you

**Comment by RussKie** [comment]
Thank you

**Pull request merged**